### DISCUSSION TITLE
Help finding an alternative to a now removed legacy method in Quaternion class.

### DISCUSSION THREAD

**Discussion by IMB11** [FabricMC/fabric-api#2383] [Mod Dev Support] [answered]
Im currently in process of updating a 1.12 mod to 1.19, its been going swimmingly smooth until I encountered this method:

`float[] Quaternion.applyPositionalRotation(float x, float y, float z)`

It seems to be removed in 1.14, does anyone have some infomation on what exactly it did and how to recreate it's functionality?

This is an example of how it was used in 1.12:

```java

Box teleportPlane = ...;

double centerX = (teleportPlane.maxX + teleportPlane.minX) / 2D;
double centerY = (teleportPlane.maxY + teleportPlane.minY) / 2D;
double centerZ = (teleportPlane.maxZ + teleportPlane.minZ) / 2D;

// newEntPos = New Entity Position
// centerY, centerZ, centerX are all calculations 
float[] appliedOffset = getQuaternion().applyPositionalRotation((float)(newEntPos.x - centerX), (float)(newEntPos.y - centerY), (float)(newEntPos.z - centerZ));
```

Thanks.

**Comment by warjort** [comment, marked as answer]
I couldn't find what you are talking about in the 1.12.2 minecraft source (might be different obfuscation mappings?).
But I did find this similar thing searching github: https://github.com/Jorch72/iChunUtil/blob/df89bc359f864b24cc65aaec1add3e4a1caa002a/src/main/java/me/ichun/mods/ichunutil/common/module/worldportals/common/portal/QuaternionFormula.java#L92

**Reply by IMB11** [reply]
Thank you so much! It's exactly what I'm looking for.

I just wanted to double check as well and it seems the mod I was updating depended on iChunUtil which makes alot of sense. 

:+1:
